src/services/iasql.ts

```diff
@@ -94,16 +95,69 @@ export async function connect(
     await conn2.query(`
       INSERT INTO iasql_module VALUES ('aws_account@0.0.1')
     `);
-    await conn2.query(`
+    if (hasCredentials) {
+      // Attach credentials
+      await attach(uid, dbAlias, dbId, awsRegion, awsAccessKeyId, awsSecretAccessKey, conn2);
+    } else {
+      let counter = 0;
+      const checkCredInterval = setInterval(async () => {
+        const updated = await maybeUpdateStatus(uid, dbAlias, dbId);
+        if (updated) clearInterval(checkCredInterval);
+        counter++;
+        if (counter === 60) clearInterval(checkCredInterval); // try for 30 min
+      }, 30000);
```

Comment:
this totally works for now, but we need two follow up tasks (not to be addressed in this PR):
1. the interval is not resilient to engine deploys. we need to move it to a graphile worker cron task. that task will also need to clean up databases that never make it to ready/active after a timeout.
2. we need to fail any iasql command that tries to talk to a pending database. also we should probably remove the `apply`/`sync` endpoints while we are at this?

Reply:
Good points, definitely